src/Diagnostics.RuntimeHost/Services/SourceWatcher/Watchers/GitHubWatcher.cs

```diff
@@ -151,8 +157,14 @@ private async Task StartWatcherInternal(bool startup)
                 LogMessage($"SourceWatcher : Start, startup = {startup.ToString()}");
                 var destDirInfo = new DirectoryInfo(_destinationCsxPath);
                 var destLastModifiedMarker = await FileHelper.GetFileContentAsync(destDirInfo.FullName, _lastModifiedMarkerName);
+                
+                if(string.IsNullOrWhiteSpace(LatestSha))
+                {
+                    LatestSha = await _githubClient.GetLatestSha();
+                }
+
+                var response = await _githubClient.GetTreeBySha(sha: LatestSha, etag: destLastModifiedMarker);
 
-                var response = await _githubClient.Get(_rootContentApiPath, etag: destLastModifiedMarker);
```

Comment:
So when we make this call, this is getting multiple shas?

Reply:
Nope this gets the full content of the github repo without truncating. 
This is the limitation of old API: https://developer.github.com/v3/repos/contents/#get-repository-content

Reply:
So now with this we don't have to keep manually removing some detectors. Nice.